src/components/structures/MatrixChat.js

```diff
@@ -1046,7 +1046,7 @@ module.exports = React.createClass({
             if (MatrixClientPeg.get().isGuest()) {
                 topBar = <GuestWarningBar />;
             }
-            else if (Notifier.supportsDesktopNotifications() && !Notifier.isEnabled() && !Notifier.isToolbarHidden()) {
+            else if (Notifier.supportsDesktopNotifications() && Notifier.shouldShowToolbar()) {
```

Comment:
I don't think this change is necessary. It's clearer to keep `isEnabled` and `isToolbarHidden` separate imho